vpc-endpoints.tf

```
@@ -1,25 +1,28 @@
 resource "aws_security_group" "sgforendpoint" {
+  count       = length(var.vpc_endpoints) != 0 ? 1 : 0
   name        = "EndpointSG"
   description = "Allow indbound and outbound traffic for VPC endpoint"
   vpc_id      = aws_vpc.main.id
 }
 
 resource "aws_security_group_rule" "allow_all_ingress" {
+  count             = length(var.vpc_endpoints) != 0 ? 1 : 0
   type              = "ingress"
   from_port         = 0
   to_port           = 0
   protocol          = "-1"
   cidr_blocks       = ["0.0.0.0/0"]
```

can we sneak this fix in too?

```suggestion
  cidr_blocks       = [var.vpc_cidr_block]
```